node_modules/.bin/js-yaml

```diff
@@ -1 +0,0 @@
-../js-yaml/bin/js-yaml.js
```

Comment:
GitHub has an Actions scaffolder / starter kit:
**[`actions/toolkit`](https://github.com/actions/toolkit#readme)**

from there I found a solid sample repo with best practices for building, linting, testing, publishing, and maintaining a GH Action:

**[`actions/javascript-action`](https://github.com/actions/javascript-action#readme)**

docs worth reading:
- [**Preparing** the code](https://github.com/actions/javascript-action#change-the-code)
- quick explainer: [**Packaging** process](https://github.com/actions/javascript-action/blob/main/README.md#package-for-distribution)
- best practices: [**Managing Published Versions**](https://github.com/actions/toolkit/blob/master/docs/action-versioning.md#recommendations).
- … and the various config boilerplate files for linting (`eslint` + `prettier`) and testing (`jest`)

<hr>

1. all the source code from `index.js` gets checked in as a single file: [`dist/index.js`](https://github.com/actions/javascript-action/blob/main/dist/index.js)
    - there are a [few other files in the `dist/` directory]((https://github.com/actions/javascript-action/tree/main/dist)):
        - there's a [`licenses.txt`](https://github.com/actions/javascript-action/blob/main/dist/licenses.txt) and
        - [some files for source maps](https://github.com/actions/javascript-action/tree/main/dist))
1. the [`action.yaml` file](https://github.com/actions/javascript-action/blob/main/action.yml#L11-L13) defines `runs:` on `dist/index.js`
    
    ```yaml
    runs:
      using: 'node16'
      main: 'dist/index.js'
    ```
    
1. the `package.json` has these dependencies:
    
    - `npm i --save-dev @vercel/ncc eslint jest`
    - `npm i --save @actions/core`
    
1. the generated `dist/index.js` (and `dist/licenses.txt`, `dist/index.js.map`, etc.) is built using [**`@vercel/ncc` bundler**](https://github.com/vercel/ncc#options) (default output directory: `dist/`). here's the npm `"scripts"`:
    
    ```json
      "main": "index.js",
      "scripts": {
        "lint": "eslint .",
        "prepare": "ncc build index.js -o dist --source-map --license licenses.txt",
        "test": "jest",
        "all": "npm run lint && npm run prepare && npm run test"
      },
    ```
    
<hr>

also, there are some example useful workflows in the `actions/javascript-action` repo:

- [`test.yml`](https://github.com/actions/javascript-action/blob/main/.github/workflows/test.yml#L16) (runs `npm ci`, `npm test`, etc.)
- [`check-dist.yml`](https://github.com/actions/javascript-action/blob/main/.github/workflows/check-dist.yml#L1-L6)
    
    ```yaml
    # `dist/index.js` is a special file in Actions.
    # When you reference an action with `uses:` in a workflow,
    # `index.js` is the code that will run.
    # For our project, we generate this file through a build process from other source files.
    # We need to make sure the checked-in `index.js` actually matches what we expect it to be.
    ```
- [`codeql-analysis.yml`](https://github.com/actions/javascript-action/blob/main/.github/workflows/codeql-analysis.yml) (for enabling GH's security/vulnerability code scanner, [CodeQL](https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/about-code-scanning-with-codeql))